src/TensorFlowNET.Core/Data/IDatasetV2.cs

```diff
@@ -4,7 +4,7 @@
 
 namespace Tensorflow
 {
-    public interface IDatasetV2 : IEnumerable<(Tensor, Tensor)>
+    public interface IDatasetV2 : IEnumerable<(Tensors, Tensors)>
```

Comment:
Can we keep both of them in the interface level?
```csharp
    public interface IDatasetV2 : IEnumerable<(Tensor, Tensor)>;
    public interface IDatasetV2 : IEnumerable<(Tensors, Tensors)>;
```

Reply:
What about keeping it implementing `IEnumerable<(Tensor, Tensor)>` and add an API to get `IEnumerator<(Tensors, Tensors)>`? If the dataset actually holds only `Tensor`, then the tennsor will be casted to `Tensors` when get the `IEnumerator<(Tensors, Tensors)>`.

However, if the dataset actually has multiple inputs as `Tensors`, then it may lead to some confusion when getting the `IEnumerable<(Tensor, Tensor)>`.